data-prepper-plugins/common/src/main/java/com/amazon/dataprepper/plugins/sink/FileSink.java

```diff
@@ -56,18 +57,31 @@ public FileSink(final String outputFile) {
     }
 
     @Override
-    public void output(Collection<Record<String>> records) {
+    public void output(Collection<Record<Object>> records) {
         try (BufferedWriter writer = Files.newBufferedWriter(Paths.get(outputFilePath),
                 StandardCharsets.UTF_8)) {
-            for (final Record<String> record : records) {
-                writer.write(record.getData());
-                writer.newLine();
+            for (final Record<Object> record : records) {
+                checkTypeAndWriteObject(record.getData(), writer);
             }
         } catch (IOException ex) {
             throw new RuntimeException(format("Encountered exception opening/creating file %s", outputFilePath), ex);
         }
     }
 
+    // Temporary function to support both trace and log ingestion pipelines.
+    // TODO: This function should be removed with the completion of: https://github.com/opensearch-project/data-prepper/issues/546
```

Comment:
Please add a task to the description of #546 to note that this needs to be done in order to complete that issue.

One task noting both this class and the unit test class should be sufficient.

Reply:
#758